### PR TITLE
chore: add function synchronous execution doc

### PR DESCRIPTION
**Pull request by altafDevRev**
## Summary
Adding the documentation for synchronous execution of functions.

## Connected Issues
- https://app.devrev.ai/devrev/works/ISS-195824

## Craftsmanship, Integrity, and Devil’s Advocacy

- [ ] Testing: Negative test cases: null or default values, crash and fault injection tests
- [ ] Testing: Boundary conditions: rolling upgrades, denial-of-service, etc.
- [ ] Testing: Fixing a few existing — flaky or permanently-broken — test cases
- [ ] Observing: Detailed error codes so machines can understand
- [ ] Observing: Adding superior metrics for future debugging
- [ ] Observing: Tracing the hairiest pathways for field serviceability
- [ ] Training: KnowledgeOps update? So AI always works.

## Story of the craft
This covers the issue mentioned here: https://app.devrev.ai/devrev/works/ISS-148474
First, we would need to add the sync supported in function. Example:
```
functions:
 - name: foobar
   description: Let me execute
   supports_synchronization: true
```
And secondly, we need to pass the sync id in the rego, along with the event_key. Example:
```
package rego
output = {"event": event, "event_key": event_key, "sync_id": sync_id} {
	input.request.method == "POST"
	event := input.request.body
	event_key := "user-enter"
	sync_id := "1234567890"
} else = {"response": response }{
	response := {
		"status_code": 400
	}
}
```

### CODE REVIEW

File: fern/docs/pages/references/functions.mdx [hunk 1]
```diff
@@ -27,6 +27,58 @@ functions:
...
+   - If supports_synchronization is true and a sync-id is provided, executions with the same sync-id are processed in order.
+   - If no sync-id is provided, executions are grouped by a random string.
+   - If supports_synchronization is false, the sync-id is ignored and executions are processed as usual.
+
+#### Platform Behavior
+
+- The platform uses the sync-id as the message group for FIFO processing.
+- This ensures that all executions with the same sync-id are processed in order, preventing race conditions or duplicate conversations.
+
+#### Use Cases
```
Comment (Atul-Butola) [inline fern/docs/pages/references/functions.mdx:76]:
```suggestion
#### Use cases
```

**Review by Atul-Butola — APPROVED** [review]

File: fern/docs/pages/references/functions.mdx [hunk 2]
```diff
@@ -27,6 +27,58 @@ functions:
 - `name`: It is the function name that should match the corresponding function name in the folder.
 - `description`: It describes the function.
 
+### Synchronous execution with synchronization key
+
+To ensure that function executions are processed in order, FIFO, for a given context such as a user or conversation, enable synchronization support in your function’s manifest and use a synchronization key, sync-id.
```
Comment (bc-devrev) [inline fern/docs/pages/references/functions.mdx:32]:
```suggestion
To ensure that function executions are processed in order, FIFO, for a given context such as a user or conversation, enable synchronization support in your function’s manifest and use a synchronization key, sync-id.

The platform uses the sync-id as the message group for FIFO processing. This ensures that all executions with the same sync-id are processed in order, preventing race conditions or duplicate conversations.

Use cases for synchronous execution include:
- Ensuring order of message processing for chatbots or messaging integrations.
- Preventing duplicate or out-of-order actions when multiple events are triggered.
```

File: fern/docs/pages/references/functions.mdx [hunk 3]
```diff
@@ -27,6 +27,58 @@ functions:
...
+
+#### How it works
```
Comment (bc-devrev) [inline fern/docs/pages/references/functions.mdx:34]: Avoid H4. By moving the content of the other information up, you don't need any subsections here.
```suggestion
```

File: fern/docs/pages/references/functions.mdx [hunk 4]
```diff
@@ -27,6 +27,58 @@ functions:
...
+
+#### Platform behavior
+
+- The platform uses the sync-id as the message group for FIFO processing.
+- This ensures that all executions with the same sync-id are processed in order, preventing race conditions or duplicate conversations.
+
+#### Use cases
+
+- Ensuring order of message processing for chatbots or messaging integrations.
+- Preventing duplicate or out-of-order actions when multiple events are triggered.
```
Comment (bc-devrev) [inline fern/docs/pages/references/functions.mdx:79]: Move to top.
```suggestion
```

**Review by bc-devrev — APPROVED** [review]
Approved with suggestions for organization.

Also, is `sync-id` correct? The code samples show `sync_id`.

**Comment by altafDevRev** [comment]
> Approved with suggestions for organization.
> 
> Also, is `sync-id` correct? The code samples show `sync_id`.

Yeah, let's make it sync_id to be consistent.
